File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/initializers/LoggerInitializer.java
package com.tinlone.startupexamplejava.initializers;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;

import com.tinlone.startupexamplejava.dependencies.Logger;

import java.util.Collections;
import java.util.List;

public class LoggerInitializer implements Initializer<Logger> {


    @NonNull
    @Override
    public Logger create(@NonNull Context context) {
        Log.i("loglog", "create: LoggerInitializer");
        Logger.initialize();
        return Logger.getInstance();
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        // 预示没有其他依赖需要初始化
        return Collections.emptyList();
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/initializers/LoggerInitializer.kt
package com.tinlone.startupexamplekotlin.initializers

import android.content.Context
import android.util.Log
import androidx.startup.Initializer
import com.tinlone.startupexamplekotlin.dependencies.Logger

class LoggerInitializer : Initializer<Logger?> {
    override fun create(context: Context): Logger {
        Log.i("loglog", "create: LoggerInitializer")
        Logger.initialize()
        return Logger.getInstance()
    }

    override fun dependencies(): List<Class<out Initializer<*>?>> {
        // 预示没有其他依赖需要初始化
        return emptyList()
    }
}